### ISSUE TITLE
"This extension may have been corrupted"

### ISSUE THREAD

**Issue by Accelyte** [KC3Kai/KC3Kai#2164]
### Versions:
Chrome Version 60.0.3112.90 (Official Build) (64-bit)
KC3 Kai ver 31.9.1 from webstore

### Steps to reproduce:
1. Updated to most recent version from version history page
2. The KC3 icon disappears
3. chrome://extensions page says extension may be corrupted. Re-enabling disables it again.

### Screenshots (if applicable):
![chrome_2017-08-14_16-02-34](https://user-images.githubusercontent.com/21363469/29263108-07374a3a-810a-11e7-8da0-fe7d63ae35ce.png)

**Comment by Accelyte** [comment]
The repair button works so the issue is closed, but I'll leave this here if there are any debugging needs...?

**Comment by Accelyte** [comment]
All settings were removed, I'm assuming chrome removed and reinstalled the extension. RIP my event info.

**Comment by sinsinpub** [comment]
RIP. it accidentally happens to chrome sometimes... only assets folder disappears for some cases. reason still unknown. wait for chrome fetching next version to replace files might fix it, if possible.

**Comment by Accelyte** [comment]
But my logs are gone permanently right? Is there any way to recover them?

**Comment by sinsinpub** [comment]
lost and unrecoverable...

It reminds us always make a backup if we care about our data. in fact, there are some ways to manually backup & restore ur chrome extension data, write one up here for future just in case:
 * Find out your Chrome Profile Path:
   * By default, if Profile directory not customized via `--user-data-dir=` command parameter, it usually can be located at:
     * Windows: `%LocalAppData%\Google\Chrome\User Data\Default`
     * Unix/Linux: `~/.config/google-chrome/default`
     * MacOS: `Users/<username>/Library/Application Support/Google/Chrome/Default`
   * Most accurate method is open `chrome://version` in your Chrome, and see the `Profile Path`.
 * If you are not using Default profile, just replace the `/Default` part with your user name.
 * Once you get your Profile Path, we will define it as `<Profile>` in  following path instructions.
 * Find our KC3Kai extension ID. For webstore users, it's `hkgmldnainaglpjngpajnnjfhpdjkohh`. For other cases, open `chrome://extensions` and find the using KC3 extension ID. we will define it as `<KC3ID>` in following path instructions.
 * Remember this two sub folders in `<Profile>` folder:
   * `Local Storage`: stores KC3 basic profile data, such as player info, fleets, map info, configs.
   * `IndexedDB`: the database stores kinds of history data.
 * Prepare your folder to put backup files at any location as you like. we will define it as `<Backup>` in following path instructions.
 * **TO BACKUP**
   * Completely close your Chrome browser first, to ensure all your data flushed into files.
   * Copy `<Profile>/Local Storage/chrome-extension_<KC3ID>_0.localstorage` and `<Profile>/Local Storage/chrome-extension_<KC3ID>_0.localstorage-journal` to `<Backup>` folder. the `-journal` file usually is 0 sized file as long as your Chrome not quits unexpectedly.
   * Copy the whole `<Profile>/IndexedDB/chrome-extension_<KC3ID>_0.indexeddb.leveldb` folder to `<Backup>`.
   * That's all.
 * **TO RESTORE**
   * Ensure your Chrome is closed, no any process is still running in background.
   * Just copy your backed up files back from `<Backup>` to the corresponding location of `<Profile>`, overwrite those files and folder (or delete them first).
   * That's all. Just remind: if you restore your old backup files backed up from old Chrome version, there is chance without luck which mess up your data thank to version changed.
 * Backup should be done before any operation has risk to make data lost, such as: reinstall extension, execute some bug fix script, or especially: if your Chrome says extension already corrupted, DO NOT click `Repair` before you backup your data. Because `Repair` is just simply uninstall (wipe out all your data) and reinstall it from webstore.

**Comment by Accelyte** [comment]
Is it possible to link KC3 to a web storage service like dropbox for cloud backup? Then KC3 can schedule a backup and restore from it if it is corrupted.

**Comment by sinsinpub** [comment]
possible, at your own risk about performance and security.

**Comment by Nya-chan** [comment]
@sinsinpub "not possible to do inside kc3, because running inside chrome sandbox cannot access ur filesystem."

You wouldn't need to access the filesystem. Just fill in dropbox address (or sth) in settings, export the backup to there regularly. If the game crashes / reinstalls, import backup manually. Is there some obvious problem with that?

**Comment by sinsinpub** [comment]
@Nya-chan the word 'backup files' i was talking about are files of chrome profile on filesystem to archive high performance backup/restore. your thinking is already done at PR #1876, building / restoring of the backup data (json string) is known be extremely time cost for long time history player.

**Comment by sinsinpub** [comment]
from Chrome 61, `Local Storage` files for KC3 cannot be backed up individually, because all data for all domains/extensions are saved into single `leveldb` folder.

**Comment by Nya-chan** [comment]
@sinsinpub "building / restoring of the backup data (json string) is known be extremely time cost for long time history player"
Couldn't this be somewhat alleviated by incremental backups? I know it's grasping straws by this point (you'd need to have all files as well as this being more difficult than other options, possibly), but it might still be something.

**Comment by Aslap** [comment]
Had this same issue today and recovered all my logs by recovering the previous version of the folder 
`C:\Users\<username>\AppData\Local\Google\Chrome\User Data\Default\IndexedDB\chrome-extension_hkgmldnainaglpjngpajnnjfhpdjkohh_0.indexeddb.leveldb`
